addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java

```diff
@@ -0,0 +1,1061 @@
+                updateFanPower(messageParts);
+
+            } else if (StringUtils.containsIgnoreCase(message, ";FAN;SPD;ACTUAL;")) {
+                updateFanSpeed(messageParts);
+
+            } else if (StringUtils.containsIgnoreCase(message, ";FAN;DIR;")) {
+                updateFanDirection(messageParts);
+
+            } else if (StringUtils.containsIgnoreCase(message, ";FAN;AUTO;")) {
+                updateFanAuto(messageParts);
+
+            } else if (StringUtils.containsIgnoreCase(message, ";FAN;WHOOSH;STATUS;")) {
+                updateFanWhoosh(messageParts);
+
+            } else if (StringUtils.containsIgnoreCase(message, ";SMARTMODE;ACTUAL;")) {
+                updateFanSmartmode(messageParts);
+
+            } else if (StringUtils.containsIgnoreCase(message, ";FAN;SPD;MIN;")) {
+                updateFanSpeedMin(messageParts);
+
+            } else if (StringUtils.containsIgnoreCase(message, ";FAN;SPD;MAX;")) {
+                updateFanSpeedMax(messageParts);
+
+            } else if (StringUtils.containsIgnoreCase(message, ";LIGHT;PWR;")) {
+                updateLightPower(messageParts);
+
+            } else if (StringUtils.containsIgnoreCase(message, ";LIGHT;LEVEL;ACTUAL;")) {
+                updateLightLevel(messageParts);
+
+            } else if (StringUtils.containsIgnoreCase(message, ";LIGHT;AUTO;")) {
+                updateLightAuto(messageParts);
+
+            } else if (StringUtils.containsIgnoreCase(message, ";SMARTERLIGHTING;")) {
+                // FIXME Find string to match on
+                updateLightSmarter(messageParts);
+
+            } else if (StringUtils.containsIgnoreCase(message, ";LIGHT;LEVEL;MIN;")) {
+                updateLightLevelMin(messageParts);
+
+            } else if (StringUtils.containsIgnoreCase(message, ";LIGHT;LEVEL;MAX;")) {
+                updateLightLevelMax(messageParts);
+
+            } else if (StringUtils.containsIgnoreCase(message, ";SNSROCC;STATUS;")) {
+                updateMotion(messageParts);
+
+            } else if (StringUtils.containsIgnoreCase(message, ";TIME;VALUE;")) {
+                updateTime(messageParts);
+
+            } else {
+                logger.debug("Received unsupported message from fan: {}", message);
+            }
+        }
+
+        private boolean isMe(String idFromDevice) {
+            // Check match on MAC address
+            if (StringUtils.equalsIgnoreCase(idFromDevice, macAddress)) {
+                return true;
+            }
+            // Didn't match MAC address, check match for label
+            if (StringUtils.equalsIgnoreCase(idFromDevice, label)) {
+                return true;
+            }
+            return false;
+        }
+
+        private void updateFanPower(String[] messageParts) {
+            logger.debug("Process fan power update (ON/OFF)");
+            if (messageParts.length != 4) {
+                logger.debug("FAN;PWR has unexpected number of parameters: {}", Arrays.toString(messageParts));
```

Comment:
This `Arrays.toString(messageParts)` conversion is always performed also when you are not debuging